src/audio/module_adapter/module/modules.c

```diff
@@ -271,16 +203,10 @@ static int modules_free(struct processing_module *mod)
 	struct comp_dev *dev = mod->dev;
 	struct module_data *md = &mod->priv;
 	struct comp_ipc_config *config = &(mod->dev->ipc_config);
-	int ret = 0;
+	int ret;
 
 	comp_info(dev, "modules_free()");
-	if (mod->is_native_sof) {
-		const struct module_interface *mod_in = mod->priv.ops;
-
-		ret = mod_in->free(mod);
-	} else {
-		ret = iadk_wrapper_free(mod->priv.module_adapter);
-	}
+	ret = iadk_wrapper_free(mod->priv.module_adapter);
```

Comment:
this return code is now overwritten below without evaluation. Maybe at least add a print if it fails.

Reply:
I will add a print. I think we would like free module code etc. regardless what its free method returns.